backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java

```diff
@@ -748,6 +774,46 @@ Optional<BPartnerComposite> getBPartnerCompositeAssertCacheHit(@NonNull final Im
 		return extractResult(bpartnerComposites);
 	}
 
+	@NonNull
+	public Optional<MetasfreshId> resolveBankAccountId(
+			@NonNull final ExternalIdentifier bankAccountIdentifier,
+			@NonNull final BPartnerComposite bPartnerComposite)
+	{
+		Check.assumeNotNull(bPartnerComposite.getOrgId(), "At this point, the orgId is always resolved on BPartnerComposite!");
+
+		return switch (bankAccountIdentifier.getType())
+				{
+					case METASFRESH_ID -> Optional.of(bankAccountIdentifier.asMetasfreshId());
+					case EXTERNAL_REFERENCE -> externalReferenceService.getJsonMetasfreshIdFromExternalReference(bPartnerComposite.getOrgId(),
+																												 bankAccountIdentifier,
+																												 BPBankAccountType.BPBankAccount)
+							.map(JsonMetasfreshId::getValue)
+							.map(MetasfreshId::of);
+					case IBAN -> bPartnerComposite.getBankAccountByIBAN(bankAccountIdentifier.asIban())
+							.map(BPartnerBankAccount::getId)
+							.map(BPartnerBankAccountId::getRepoId)
+							.map(MetasfreshId::of);
+					case QR_IBAN -> bPartnerComposite.getBankAccountByQrIban(bankAccountIdentifier.asQrIban())
+							.map(BPartnerBankAccount::getId)
+							.map(BPartnerBankAccountId::getRepoId)
```

Comment:
imho that's not needed because we have MetasfreshId#of(de.metas.util.lang.RepoIdAware)

Reply:
(same thing a couple of lines above)